apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,325 @@
+                    <Text variant="body" color="gray-80" weight="medium">
+                        Estimated Gas Fees
+                    </Text>
+                    <div className="text-gray-60 h-4 items-end flex">
+                        <IconTooltip tip="Estimated Gas Fees" placement="top" />
+                    </div>
+                </div>
+                <Text variant="body" color="gray-90" weight="medium">
+                    {formattedGas} {gasSymbol}
+                </Text>
+            </div>
+        </Loading>
+    );
+}
+
+// Set the initial gasEstimation from initial amount
+// base on the input amount feild update the gasEstimation value
+// Seperating the gasEstimation from the formik context to access the input amount value and update the gasEstimation value
+export function SendTokenForm({
+    coinType,
+    onSubmit,
+    initialAmount = '',
+    initialTo = '',
+    initialGasEstimation = 0,
+}: SendTokenFormProps) {
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const coins = allCoins.filter(
+        ({ type }) => CoinAPI.getCoinType(type) === coinType
+    );
+
+    const suiCoins = allCoins.filter(
+        ({ type }) => CoinAPI.getCoinType(type) === SUI_TYPE_ARG
+    );
+
+    const gasAggregateBalance = aggregateBalances[SUI_TYPE_ARG] || BigInt(0);
+    const coinSymbol = (coinType && CoinAPI.getCoinSymbol(coinType)) || '';
+    const [coinDecimals] = useCoinDecimals(coinType);
+    const [gasDecimals, gasQueryResult] = useCoinDecimals(SUI_TYPE_ARG);
+    const maxSuiSingleCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
+
+    const validationSchemaStepOne = useMemo(
+        () =>
+            createValidationSchemaStepOne(
+                coinType || '',
+                coinBalance,
+                coinSymbol,
+                gasAggregateBalance,
+                coinDecimals,
+                gasDecimals,
+                initialGasEstimation,
+                maxSuiSingleCoinBalance
+            ),
+        [
+            coinType,
+            coinBalance,
+            coinSymbol,
+            gasAggregateBalance,
+            coinDecimals,
+            gasDecimals,
+            initialGasEstimation,
+            maxSuiSingleCoinBalance,
+        ]
+    );
+
+    const [tokenBalance, symbol, queryResult] = useFormatCoin(
+        coinBalance,
+        coinType,
+        CoinFormat.FULL
+    );
+
+    // remove the comma from the token balance
+    const formattedTokenBalance = tokenBalance.replace(/,/g, '');
+    const maxToken = parseAmount(formattedTokenBalance, coinDecimals);
+    const initAmountBig = parseAmount(initialAmount, coinDecimals);
+
+    return (
+        <Loading loading={queryResult.isLoading || gasQueryResult.isLoading}>
```

Comment:
let's add in loading the coinDecimals loading value. It's probably in the queryResult but lets add it